Detect libcanberra with CMake, not pkg-config.

The latter does not produce full paths for libraries, leading to build failures
if -L/path/to is not passed to the linker as well.
--- phonon/kcm/CMakeLists.txt
+++ phonon/kcm/CMakeLists.txt
@@ -11,7 +11,11 @@ set_package_properties(GLIB2 PROPERTIES DESCRIPTION "Low-level core library for
                        TYPE OPTIONAL
                       )
 
-pkg_check_modules(CANBERRA libcanberra)
+find_path(CANBERRA_INCLUDE_DIRS canberra.h)
+find_library(CANBERRA_LIBRARIES canberra)
+if (CANBERRA_INCLUDE_DIRS AND CANBERRA_LIBRARIES)
+  set(CANBERRA_FOUND TRUE)
+endif ()
 add_feature_info ("Audio setup GUI" CANBERRA_FOUND
                   "libcanberra is needed for audio setup GUI
                   * http://0pointer.de/lennart/projects/libcanberra"
